src/main/java/net/coderbot/iris/pipeline/ShadowRenderer.java

```diff
@@ -484,16 +380,15 @@ public void renderShadows(LevelRendererAccessor levelRenderer, Camera playerCame
 		// Set up the viewport
 		RenderSystem.viewport(0, 0, resolution, resolution);
 
-		// Set up our orthographic projection matrix and load it into the legacy matrix stack
-		FloatBuffer projMatBuf = FloatBuffer.allocate(16);
-		projMatBuf.put(orthoMatrix);
-		projMatBuf.flip();
+		// Set up our orthographic projection matrix
+		float[] orthoMatrix = ShadowMatrices.createOrthoMatrix(halfPlaneLength);
 
-		Matrix4f projectionMatrix = new Matrix4f();
-		projectionMatrix.load(projMatBuf);
+		Matrix4f projMatrix = new Matrix4f();
+		((Matrix4fAccess) (Object) projMatrix).copyFromArray(orthoMatrix);
 
-		Matrix4f previousProjectionMatrix = RenderSystem.getProjectionMatrix();
-		RenderSystem.setProjectionMatrix(projectionMatrix);
+		// Load projection matrix into RenderSystem and backup the old one
+		RenderSystem.backupProjectionMatrix();
```

Comment:
We should probably just save the projection matrix in our own variable in case something else tries to call backupProjectionMatrix in shadow rendering